src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/IntegralMutatorFactory.java

```diff
@@ -289,8 +289,26 @@ protected final long mutateImpl(long value, PseudoRandom prng) {
       return value;
     }
 
-    protected final long crossOverImpl(long value, long otherValue, PseudoRandom prng) {
-      return value;
+    protected final long crossOverImpl(long x, long y, PseudoRandom prng) {
+      switch (prng.indexIn(4)) {
+        case 0:
+          return mean(x, y);
+        case 1:
+          return forceInRange(x | y);
+        case 2:
+          return forceInRange(x & y);
+        case 3:
+          return forceInRange(x ^ y);
+        default:
+          throw new AssertionError("Invalid cross over function.");
+      }
+    }
```

Comment:
For another cross-over possibility, we could add another bitwise mix of `x` and `y`, controlled by a random mask. This approach combines parts of `x` and `y` at a finer granularity than the existing methods. Something like:

```java
  case 4:
            long mask = prng.closedRange(0, Integer.MAX_VALUE);
            return forceInRange((x & mask) | (y & ~mask));
```

WDYT?

Reply:
Looks good. I would wait with more sophisticated cross-overs until we have benchmarks, though.